Firestore/core/src/index/firestore_index_value_writer.cc

```diff
@@ -105,6 +107,31 @@ void WriteIndexArray(const google_firestore_v1_ArrayValue& array_index_value,
   }
 }
 
+void WriteIndexVector(const google_firestore_v1_MapValue& map_index_value,
+                      DirectionalIndexByteEncoder* encoder) {
+  WriteValueTypeLabel(encoder, IndexType::kVector);
+
+  absl::optional<pb_size_t> valueIndex =
+      model::IndexOfKey(map_index_value, model::kRawVectorValueFieldKey,
+                        model::kVectorValueFieldKey);
+
+  if (!valueIndex.has_value() ||
+      map_index_value.fields[valueIndex.value()].value.which_value_type !=
+          google_firestore_v1_Value_array_value_tag) {
```

Comment:
should have been `value_index`. C++ files should follow the Google [style guide](https://google.github.io/styleguide/cppguide.html#Variable_Names). There's a mix of snake case and camel case here 😵